scipy/stats/_continuous_distns.py

```diff
@@ -6121,15 +6122,88 @@ def _entropy(self, c):
+                    return (data.shape[0] /
+                            np.sum(np.log((data - location) / scale)))
+                else:
+                    # if `fshape` is fixed, return that instead.
+                    return fshape
+
+            def get_location(scale):
+                # Return the numerically optimal location.
+                return np.min(data) - scale
+
+            def dLdScale(shape, scale):
+                # The partial derivative of the log-likelihood function w.r.t.
+                # the scale.
+                return data.shape[0] * shape / scale
+
+            def dLdLocation(shape, location):
+                # The partial derivative of the log-likelihood function w.r.t.
+                # the location.
+                return (shape + 1) * np.sum(1 / (data - location))
+
+            def fun_to_solve(scale):
+                # optimize over the scale, setting the partial derivatives
+                # w.r.t. to location and scale equal.
+                location = get_location(scale)
+                shape = get_shape(scale, location)
+                return dLdLocation(shape, location) - dLdScale(shape, scale)
+
+            def interval_contains_root(lbrack, rbrack):
+                # returns true if the signs disagree.
+                return (np.sign(fun_to_solve(lbrack)) !=
+                        np.sign(fun_to_solve(rbrack)))
+            # set brackets for `root_scalar` to use when optimizing over the
+            # scale such that a root is likely between them.
+            lbrack, rbrack = .1, np.max(data)
+            # if a root is not between the brackets, iteratively expand them
+            # until they include a sign change, checking after each bracket is
+            # modified.
+            if not interval_contains_root(lbrack, rbrack):
+                while True:
+                    lbrack /= 2
+                    if interval_contains_root(lbrack, rbrack):
+                        break
+                    rbrack *= 2
+                    if interval_contains_root(lbrack, rbrack):
+                        break
+            res = root_scalar(fun_to_solve, bracket=[lbrack, rbrack])
+
+            if res.converged:
+                fscale = res.root
+                floc = get_location(fscale)
+                fshape = get_shape(fscale, floc)
+
+                # The Pareto distribution requires that its parameters satisfy
+                # the condition `fscale + floc <= min(data)`. If this condition
+                # is not satisfied, reduce the scale with `np.nextafter` to
+                # ensure that data does not fall outside of the support.
+                if not ((fscale + floc) <= np.min(data)):
+                    fscale = np.nextafter(fscale, 0)
+                return fshape, floc, fscale
+            else:
+                return super().fit(data, **kwds)
+        elif floc is None:
+            # fitting free `floc` requires free `fscale`.
```

Comment:
I think `loc = np.min(data) - scale` works as the best fit here, too.

Reply:
Where did this come from?
>  # fitting free `floc` requires free `fscale`.

Did I write that sort of thing somewhere? I'm pretty sure we always want to maximize both `loc` and `scale` subject to the constraint that `loc + scale <= np.min(data)`, regardless of which one is given.

Reply:
Implemented